lib/resources/docker.rb

```diff
@@ -232,5 +257,17 @@ def parse_images
       warn 'Could not parse `docker images` output'
       []
     end
+
+    def parse_plugins
+      plugins = inspec.command('docker plugin ls --format \'{"id": {{json .ID}}, "name": "{{ with split .Name ":"}}{{index . 0}}{{end}}", "version": "{{ with split .Name ":"}}{{index . 1}}{{end}}", "enabled": {{json .Enabled}} }\'').stdout
+      c_plugins = []
+      plugins.each_line { |entry|
+        c_plugins.push(JSON.parse(entry))
+      }
+      c_plugins
+    rescue JSON::ParserError => _e
+      warn 'Could not parse `docker plugin ls` output'
```

Comment:
This cannot be handled as we are only getting stdout (https://github.com/inspec/inspec/pull/3074/files#diff-a405d35eca4d46fa85c19029d6f3b79bR259) and any errors always returns a valid JSON. This would need a lot of code change in the docker resource to handle properly.

Reply:
Can't we just do `fail_resource('Could not parse `docker plugin ls` output')` ?

Reply:
I think we should do ^ also.

If you can't parse the JSON then we should fail the resource. Granted, I see your point if we get a legitimate error in valid JSON.